venues/reproducibility-challenge.github.io/Reproducibility_Challenge/NeurIPS/2019/python/notify.py

```diff
@@ -0,0 +1,80 @@
+
+import argparse
+import openreview
+from openreview import tools
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+parser.add_argument('--freq', required=True, help="Daily or Weekly")
+parser.add_argument('--tcdate', required=True, help="Time in msec of last update")
+parser.add_argument('--notify_author', default=False)
+args = parser.parse_args()
+
+client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+print("Connected to "+client.baseurl)
+conference_id = 'NeurIPS.cc/2019/Reproducibility_Challenge'
+
+def get_author_id(forumNote):
+    # get submission author and add to email list if author hasn't set a notification tag
+    profile = None
+    # if there is a primary author email, and it is not on the list already, add it
+    author_email = next(s for s in forumNote.content['authorids'] if s)
+    if author_email:
+        try:
+            profile = client.get_profile(forumNote.content['authorids'][0])
```

Comment:
don't need to get the profile. the messages endpoint finds the profile of each recipient to get the preferredEmail

Reply:
Sorry, you are using this to get the signature of the tag.